marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/entities/search/impl/CreatedOnFacetHandlerTest.java

```diff
@@ -32,10 +41,73 @@ public void testCreatedOnDateRangeFacet() {
         Fragment fragment = toFragment(queryDefinition);
 
         assertEquals(Arrays.asList(constraintName, constraintName), fragment.getElementValues("//search:constraint-name"));
-        assertTrue(fragment.getElementValue("//search:value[../search:range-operator='GE']").contains(lowerBound),
-                "Expected lower bound to include: " + lowerBound);
+        assertTrue(fragment.getElementValue("//search:value[../search:range-operator='GE']").contains(expectedLowerBound),
+                "Expected lower bound to include: " + expectedLowerBound);
         assertTrue(fragment.getElementValue("//search:value[../search:range-operator='LT']").contains(expectedUpperBound),
                 "Upper bound is one day more than what was passed in order to include the passed in date as well" +
                         " for comparison. Expected upper bound to include: " + expectedUpperBound);
     }
+
+    @Test
+    public void testComputeDateRangeTimeWindows() {
+        String zoneOffset = "-420";
+        ZoneId zoneId = ZoneId.ofOffset("UTC", ZoneOffset.ofTotalSeconds(Integer.parseInt(zoneOffset)*60));
+        DocSearchQueryInfo.FacetData facetData = new DocSearchQueryInfo.FacetData();
+        DocSearchQueryInfo.RangeValues rangeValues = new DocSearchQueryInfo.RangeValues();
+
+        // For time window Today.
```

Comment:
Because you have 4 cases in your switch in the implementation, I'd expect at least 4 separate test methods. And because of the presence of "if" and "try" blocks for "Custom", I'd expect multiple test methods for that case, one for each scenario. This will be much more readable than one long test that covers different scenarios, and it'll be easier to debug if/when one of the scenarios breaks.